sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -487,20 +487,28 @@ string_class ProgramManager::getProgramBuildLog(const RT::PiProgram &Program,
   string_class Log = "The program was built for " +
                      std::to_string(PIDevices.size()) + " devices";
   for (RT::PiDevice &Device : PIDevices) {
+    std::string DeviceBuildInfoString;
     Plugin.call<PiApiKind::piProgramGetBuildInfo>(
         Program, Device, CL_PROGRAM_BUILD_LOG, 0, nullptr, &Size);
-    vector_class<char> DeviceBuildInfo(Size);
-    Plugin.call<PiApiKind::piProgramGetBuildInfo>(
-        Program, Device, CL_PROGRAM_BUILD_LOG, Size, DeviceBuildInfo.data(),
-        nullptr);
+    if (Size > 0) {
+      vector_class<char> DeviceBuildInfo(Size);
+      Plugin.call<PiApiKind::piProgramGetBuildInfo>(
+          Program, Device, CL_PROGRAM_BUILD_LOG, Size, DeviceBuildInfo.data(),
+          nullptr);
+      DeviceBuildInfoString = std::string(DeviceBuildInfo.data());
```

Comment:
Just looking at this code again...
Are you sure `DeviceBuildInfo.data()` is `\0` terminated?
I have a feeling that it should be also
```c++
vector_class<char> DeviceBuildInfo(Size + 1);
```

Reply:
Rereading again... :)
So `DeviceBuildInfo` is initialized with `Size` `\0`, this can work only if `Plugin.call` write at most `Size` characters including the trailing `\0`... in that case yes it can work...

Reply:
Thanks. Resolving this conversation.